sanic/mixins/routes.py

```diff
@@ -702,6 +704,8 @@ def static(
         content_type: Optional[bool] = None,
         apply: bool = True,
         resource_type: Optional[str] = None,
+        autoindex: bool = False,
+        index: Optional[Union[str, Sequence[str]]] = None,
```

Comment:
If we are going with these two separate arguments, better name autoindex to something that does NOT get mixed with the other argument. Now one would assume that `autoindex=True` is needed for `index` to have an effect.

Maybe the whole thing could be named FolderViewer, FileListing, FileExplorer or something alike, because it doesn't actually view any files and isn't a (web) browser, and again that sort of naming would be fully self-explanatory.

Out of those that I could think of, I might prefer `FileListing` for the class, `response.file_listing` and simply `listings` as the kwarg here.

If you wish to avoid the renaming in other places, at least change the argument to `browser`, to match `response.file_browser` and to avoid the aforementioned confusion between the two kwargs (maybe you already suggested this name in an earlier thread).

The possibility for multiple index files is very much unnecessary (everyone only ever want index.html, if anything). I added it as a side product of the single argument already requiring a tuple. Removing the sequence option would simplify typing and documentation, even if it only saves a couple of lines in implementation.

Reply:
> Now one would assume that autoindex=True is needed for index to have an effect.
It's not. And, if the current implementation requires it then that's a mistake that will be fixed in testing.

As for naming, autoindex is quite literally what's in use. I don't disagree that it is misleading in the larger scheme though. Rather, it's parity with nginx that drove that naming.

As for single v multiple, it seems like a familiar option for a couple lines of code. It totally feels like something that someone would ask for. Not tied to keeping it, but I also don't see it as a burden to maintain.

Reply:
I mean it is misleading **locally**, with app.static having two adjacent arguments index arguments that intuitively *seem* like the first one enables enables the second one.

Since for that reason `autoindex` needs to be changed, we might as well be consistent with how the file browser feature is called all around Sanic, and only use one name for it (suggestions above, but file browser is also OK, now that autoindex is not available).

Reply:
How about `directory_view=True`. I would like to avoid "folder" because I do not think we use that term anywhere.

Reply:
It's up to you, possibly also dirview or other abbreviation, but rename the response and other parts accordingly (there is a project-wide rename function in VS Code that helps but it misses docstrings/comments and those chains of passing the same thing through other functions, so global find/replace could also be helpful).

TLDR: same name for all.